src/engine/TextIndexScanForEntity.h

```diff
@@ -9,56 +13,71 @@
 // The entities are saved to the entityVar_. If the operation is called on a
 // fixed entity instead, it only returns entries that contain this entity.
 class TextIndexScanForEntity : public Operation {
- public:
+ private:
   struct VarOrFixedEntity {
     VarOrFixedEntity(const QueryExecutionContext* qec,
-                     const std::variant<Variable, std::string>& entity) {
+                     std::variant<Variable, std::string> entity) {
       if (std::holds_alternative<std::string>(entity)) {
-        fixedEntity_.emplace(std::get<std::string>(entity));
-        bool success =
-            qec->getIndex().getVocab().getId(fixedEntity_.value(), &index_);
+        VocabIndex index;
+        std::string fixedEntity = std::move(*std::get_if<std::string>(&entity));
+        bool success = qec->getIndex().getVocab().getId(fixedEntity, &index);
         if (!success) {
           throw std::runtime_error(
-              "The entity " + fixedEntity_.value() +
+              "The entity " + fixedEntity +
               " is not part of the underlying knowledge graph and can "
-              "therefore "
-              "not be used as the object of ql:contains-entity");
+              "therefore not be used as the object of ql:contains-entity");
         }
+        entity_ = std::pair<std::string, VocabIndex>(std::move(fixedEntity),
+                                                     std::move(index));
       } else {
-        entityVar_.emplace(std::get<Variable>(entity));
+        entity_ = std::move(*std::get_if<Variable>(&entity));
       }
     }
     ~VarOrFixedEntity() = default;
 
-    std::optional<std::string> fixedEntity_ = std::nullopt;
-    VocabIndex index_;
-    std::optional<Variable> entityVar_ = std::nullopt;
+    bool hasFixedEntity() const {
+      return std::holds_alternative<std::pair<std::string, VocabIndex>>(
```

Comment:
early on define a `using FixedEntity = std::pair<...>` in this class, then the typing gets less verbose.